packages/adblocker/src/engine/engine.ts

```diff
@@ -828,6 +917,38 @@ export default class FilterEngine extends EventEmitter<
       isFilterExcluded: this.isFilterExcluded.bind(this),
     });
 
+    if (this.hasListeners()) {
+      const context: CosmeticFilterMatchingContext = {
+        url,
+        domain,
+        hostname,
+
+        classes,
+        hrefs,
+        ids,
+
+        allowGenericHides,
+        allowSpecificHides,
+
+        getBaseRules,
+        getInjectionRules,
+        getExtendedRules,
+        getRulesFromDOM,
+        getRulesFromHostname,
```

Comment:
Nit. Would it make sense to reuse part of the object we already create in the arguments for the function call above instead of duplicating?

Reply:
like this?
```js
const context: CosmeticFilterMatchingContext = {
   ...arguments[0],
   isFilterExcluded: undefined,
   filterType: FilterType.COSMETIC,
};
```